pkg/prometheus/rule_configmap_migration.go

```diff
@@ -97,10 +97,12 @@ func (c *Operator) getRuleCMs(ns string, cmLabelSelector *metav1.LabelSelector)
 	}
 
 	configMaps := []*v1.ConfigMap{}
-
 	err = cache.ListAllByNamespace(c.cmapInf.GetIndexer(), ns, cmSelector, func(obj interface{}) {
 		configMaps = append(configMaps, obj.(*v1.ConfigMap))
 	})
+	if err != nil {
```

Comment:
this was an actual bug; we were ignoring the error returned by `cache.ListAllByNamespace`